app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -75,13 +75,8 @@ def batch_callbacks(batch)
   end
 
   def delete_associations(object)
-    object_associations = associations_to_destroy_for(object)
-    association_names = object_associations.map(&:name)
-    preloaded_object = association_names.any? ? object.class.eager_load(*association_names).find(object.id) : object
-
-    object_associations.each do |reflection|
-      worker = association_delete_worker(reflection)
-      delete_objects_of_association(preloaded_object, reflection.name, worker)
+    object.class.reflect_on_all_associations.each do |reflection|
```

Comment:
This will also return `belong_to` associations which will later be rejected because of https://github.com/3scale/porta/blob/d8dc41e0389b370b62a90126fcd14116cf742781/app/workers/delete_object_hierarchy_worker.rb#L97. Can we only created `ReflectionDestroyer` objects for the associations that really matter?

Reply:
Update: `dependent: destroy` can be in `belongs_to` when used in conjunction with `has_one`. (It can also be when in conjunction with `has_many` although expressly recommended otherwise in the Rails documentation.)

For example:
```ruby
class Parent < ApplicationRecord
  has_one :child
end

class Child < ApplicationRecord
  belongs_to :parent, dependent: :destroy
end
```

Will deleting the `Child` cause to try to delete the `Parent` too?

Reply:
I don't know, but that is what `dependent: :destroy` means so it definitely should 🤔But I will check 👍